auth/middleware_test.go

```diff
@@ -36,7 +38,16 @@ func TestEnd2End(t *testing.T) {
 				Audience("notMyClient", oidcMockServer.Config.ClientID).
 				Build(),
 			wantErr: false,
-		}, {
+		},
+		{
+			name:   "valid with single aud",
```

Comment:
azp is not implemented in OIDC struct and I believe, for azp validation in case aud is missing we would need to provide custom validator or set azp value to aud before validation takes place